Tasks/PerformActionOnMachineGroup/AzureHelper.ps1

```diff
@@ -60,6 +60,32 @@ function Restart-MachineInProvider
 
 function Initialize-AzureHelper
 {
+    Write-Verbose "Entering in azure-initializer" -Verbose
+
     Switch-AzureMode AzureResourceManager
-    Write-Verbose "Switched to AzureResourceManager" -Verbose
+    if($machineGroup.ProviderDataList.Count -gt 0)
+    {
+        $providerDataName = $machineGroup.ProviderDataList[0].Name
+        Write-Verbose "ProviderDataName : $providerDataName" -Verbose
+        $providerData = Get-ProviderData -ProviderDataName $providerDataName -Connection $connection
+        $subscriptionName = $providerData.Properties.GetProperty("SubscriptionName")     
```

Comment:
these variables are associated with providerdatalist only that's why I put them here only